### DISCUSSION TITLE
Is there a variable where I can find all defined colors?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#8944] [General - Components / Directives / etc] [answered]
I would like to pick a random color defined in Quasar. Is there a variable that holds them all?

**Comment by metalsadman** [comment, marked as answer]
> I would like to pick a random color defined in Quasar. Is there a variable that holds them all?

https://quasar.dev/style/sass-scss-variables#variables-list

**Reply by wsw70** [reply]
Thanks, but is there a way to programmatically access them? Something along the lines of 

    randomColor = arrayOfAllColors[Math.floor(Math.random() * arrayOfAllColors.length)]

**Reply by metalsadman** [reply]
> Thanks, but is there a way to programmatically access them? Something along the lines of
> 
> ```
> randomColor = arrayOfAllColors[Math.floor(Math.random() * arrayOfAllColors.length)]
> ```

They are suffixed with numbers as you can see in the variable list, so you can randomize those, as for the names then you'll have to somehow make an array for those.

Also take a look in to color utils if you want to extract the color values of those variables in js https://quasar.dev/quasar-utils/color-utils.

**Reply by wsw70** [reply]
Thanks for the color utils hint, I am trying to register new colors but this fails (I will open a separate question for that)
